### PR TITLE
Issue #450: Add new sidebar to edit page

### LINKED ISSUE #450: Add new side bar to edit page
Eventually, the edit experience will be more wizard like, but for now, we are just going to tackle changing the side bar, and have it jump to the different sections of the existing edit page when clicked. 

@alexanderturinske We can probably keep the buttons to add a new service where it exists in the current edit page and not add it to the sidebar for now. Also, programs don't exist yet, so that section can be ignored.

image

### PR DESCRIPTION
**Pull request by alexanderturinske**
- Creates sidebar component
 - Closes #450

### CODE REVIEW

File: app/components/Edit/EditSections.js [hunk 1]
```diff
@@ -138,25 +136,23 @@ function postNotes(notesObj, promises, uriObj) {
       if (key < 0) {
         const uri = `/api/${uriObj.path}/${uriObj.id}/notes`;
         promises.push(dataService.post(uri, { note: currentNote }));
+      } else if (currentNote.isRemoved) {
+        const uri = `/api/notes/${key}`;
+        promises.push(dataService.APIDelete(uri));
```
Comment (alexanderturinske) [inline app/components/Edit/EditSections.js]: I have VSCode auto lint/beautify so it did some refactoring. I looked through the changes and trust it.

File: app/components/Resource/Resource.js [hunk 2]
```diff
@@ -52,21 +51,6 @@ class Resource extends Component {
       });
   }
 
-  hapCertify() {
```
Comment (alexanderturinske) [inline app/components/Resource/Resource.js]: This was the HAP method being used

File: app/components/Edit/EditSections.js [hunk 3]
```diff
@@ -561,16 +553,22 @@ class EditSections extends React.Component {
   }
 
   certifyHAP() {
```
Comment (alexanderturinske) [inline app/components/Edit/EditSections.js]: This method wasn't being used. The method being used to certifyHAP was called HAPcertify and was in Resource.js, but I moved it here.

File: app/components/Edit/EditSections.js [hunk 4]
```diff
@@ -657,63 +655,52 @@ class EditSections extends React.Component {
...
-      <button className="edit--aside--content--button" key='submit' disabled={this.state.submitting} onClick={this.handleSubmit}>Save Changes</button>,
-      <button className="edit--aside--content--button cancel--button" key='cancel' onClick={this.handleCancel}>Discard Changes</button>,
-      <button className="edit--aside--content--deactivate" key='deactive' disabled={this.state.submitting} onClick={() => this.handleDeactivation('resource', resource.id)}>Deactivate</button>
-    ];
-    if (this.state.newResource) {
-      actionButtons = [
-        <button className="edit--aside--content--button" key='submit' disabled={this.state.submitting} onClick={this.createResource}>Submit</button>,
-        <button className="edit--aside--content--button cancel--button" key='cancel' onClick={this.handleCancel}>Cancel</button>,
-      ];
-    }
```
Comment (alexanderturinske) [inline app/components/Edit/EditSections.js]: All moved into the sidebar

File: app/components/Edit/EditSidebar.jsx [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        >Cancel</button>,
+      ];
+    }
+    if (!resource.certified) {
+      actionButtons.push(
+        <button
+          className="edit--aside--content--button hap--button"
+          key="hap"
+          onClick={this.props.certifyHAP}
+        >HAP Approve</button>,
```
Comment (alexanderturinske) [inline app/components/Edit/EditSidebar.jsx]: Lots of copy and paste logic

File: app/components/Edit/EditSidebar.jsx [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    return (
+      <nav className="sidebar">
+        <ul className="sidebar--org--info">
+          <li className="sidebar--list--heading"><h3>Organization</h3></li>
+          <li className="sidebar--organization--name">{resource.name}</li>
+        </ul>
+        <ul>
+          <hr />
+          <li className="sidebar--list--heading"><h3><a href="#services">Services</a></h3></li>
+          {resource.services.map(service => (
```
Comment (lexholden) [inline app/components/Edit/EditSidebar.jsx]: For useability, these could be anchor tags that link to the top of that individual service section on the page.

I notice this doesn't update as I add a new service, which I guess would also be useful (serviceNameField.onblur updates the list state?)

File: app/components/Edit/EditSidebar.jsx [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+          key="hap"
+          onClick={this.props.certifyHAP}
+        >HAP Approve</button>,
+      );
+    }
+    return (
+      <nav className="sidebar">
+        <ul className="sidebar--org--info">
+          <li className="sidebar--list--heading"><h3>Organization</h3></li>
+          <li className="sidebar--organization--name">{resource.name}</li>
```
Comment (lexholden) [inline app/components/Edit/EditSidebar.jsx]: It's a bit confusing that this class is coloured blue, and so looks like a link, when it's not.

File: app/components/Edit/EditSidebar.jsx [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+  render() {
+    const resource = this.props.resource;
+    let actionButtons = [
+      <button
+        className="edit--aside--content--button"
+        key="submit"
+        disabled={this.props.submitting}
+        onClick={this.props.handleSubmit}
+      >Save Changes</button>,
+      <button
```
Comment (lexholden) [inline app/components/Edit/EditSidebar.jsx]: Trivial point, with our button styles more than anything, but on narrow screens this overflows messily: 

screenshot 2018-03-25 00 46 49 2

Creating #453 to track in general as a followup

File: app/components/Edit/EditSidebar.jsx (hunk 6 above)
Comment (alexanderturinske) [reply app/components/Edit/EditSidebar.jsx]: Thanks for the feedback. Will do.

File: app/components/Edit/EditSidebar.jsx (hunk 7 above)
Comment (alexanderturinske) [reply app/components/Edit/EditSidebar.jsx]: The mockup shows this as blue (https://github.com/ShelterTechSF/askdarcel-web/issues/450)

File: app/components/Edit/EditSidebar.jsx (hunk 8 above)
Comment (alexanderturinske) [reply app/components/Edit/EditSidebar.jsx]: Nice. Thanks!

**Comment by alexanderturinske** [comment]
@lexholden Addressed the being able to click on services in the sidebar and new services not being added to the sidebar.

**Comment by lexholden** [comment]
Awesome. Merging in.

**Comment by richardxia** [comment]
Looks like this PR broke the "Add new resource" page.

```
EditSidebar.jsx:67 Uncaught TypeError: Cannot read property 'map' of undefined
    at EditSidebar.render (EditSidebar.jsx:67)
    at ReactCompositeComponent.js:795
    at measureLifeCyclePerf (ReactCompositeComponent.js:75)
    at ReactCompositeComponentWrapper._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:794)
    at ReactCompositeComponentWrapper._renderValidatedComponent (ReactCompositeComponent.js:821)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:361)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:257)
    at Object.mountComponent (ReactReconciler.js:45)
    at ReactCompositeComponentWrapper.performInitialMount (ReactCompositeComponent.js:370)
    at ReactCompositeComponentWrapper.mountComponent (ReactCompositeComponent.js:257)
```

Ironically, I discovered this as I was about to write a TestCafe test for the Add new resource page.

**Comment by lexholden** [comment]
Hmm, whoops. I thought I tested it thoroughly for my first review but did not for the minor subsequent change. Investigating, though perfect example of the importance of upping our coverage ;)

**Comment by alexanderturinske** [comment]
Just saw all of this. I forgot to test adding a new resource as well; definitely my bad. +1 to updating test coverage

File: app/components/Edit/EditSidebar.jsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        onClick={this.props.handleCancel}
+      >Discard Changes</button>,
+      <button
+        className="edit--aside--content--button deactivate--button"
+        key="deactive"
+        disabled={this.props.submitting}
+        onClick={() => this.props.handleDeactivation('resource', resource.id)}
+      >Deactivate</button>,
+    ];
+    if (this.props.newResource) {
```
Comment (alexanderturinske) [inline app/components/Edit/EditSidebar.jsx]: For some reason, this expression isn't evaluating correctly.

**Pull request merged**